cmds/core/wget/wget_test.go

```diff
@@ -40,120 +43,108 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-var tests = []struct {
-	name    string
-	flags   []string // in, %[1]d is the server's port, %[2] is an unopen port
-	url     string   // in
-	content string   // out
-	retCode int      // out
-}{
-	{
-		name:    "basic",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	{
-		name:    "ipv4",
-		flags:   []string{},
-		url:     "http://127.0.0.1:%[1]d/200",
-		content: content,
-		retCode: 0,
-	},
-	// TODO: CircleCI does not support ipv6
-	// {
-	// 	name:    "ipv6",
-	// 	flags:   []string{},
-	// 	url:     "http://[::1]:%[1]d/200",
-	// 	content: content,
-	// 	retCode: 0,
-	// },
-	{
-		name:    "redirect",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/302",
-		content: "",
-		retCode: 0,
-	},
-	{
-		name:    "4xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/404",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "5xx error",
-		flags:   []string{},
-		url:     "http://localhost:%[1]d/500",
-		content: "",
-		retCode: 1,
-	},
-	{
-		name:    "no server",
-		flags:   []string{},
-		url:     "http://localhost:%[2]d/200",
-		content: "",
-		retCode: 1,
-	},
-}
-
-func getListener(t *testing.T) (net.Listener, int) {
-	t.Helper()
-	l, err := net.Listen("tcp", ":0")
+func getUnusedPort() (int, error) {
+	a, err := net.ResolveTCPAddr("tcp", ":0")
 	if err != nil {
-		t.Fatalf("error setting up TCP listener: %v", err)
+		return 0, err
 	}
-	return l, l.Addr().(*net.TCPAddr).Port
+	l, err := net.ListenTCP("tcp", a)
+	if err != nil {
+		return 0, err
+	}
+
+	l.Close()
```

Comment:
IIUC this may not work as intended. l.Close() will close the listening port, so the port number *may* be reused by another test/process (in a race condition).   The old code relies on the caller of `getUnusedPort` to defer the port closing when the test is done... and it also sets up a fake TCP listener on that port which immediately closes the connection (you'll probably have to change the test to expect something other than `syscall.ECONNREFUSED`).

I think it makes sense to copy that here, WDYT?

Reply:
You are absolutely right, ports can be reused. I probably was confused by test name "no server". So I revert to previous approach of closing connections and checking for io.EOF error.